src/minecraftTransform.js

```diff
@@ -1,62 +1,57 @@
 import { Transform } from 'stream'
-import MapColor from './mapColors'
+import { fromImage } from './mapColors'
+import cwise from 'cwise'
+import ndarray from 'ndarray'
 
 export class ComputeColors extends Transform {
-	// buffer to id array
-	_transform(chunk, _, next) {
-		MapColor.fromImage(chunk)
-			.then(this.push)
-			.then(next)
-	}
-}
-
-export class ComputeKnown extends Transform {
-	constructor(width, height) {
+	constructor() {
 		super({ readableObjectMode: true })
-		Object.assign(this, { width, height })
-		this.last = new Uint8Array(width * height)
 	}
 
-	_transform(chunk, _, next) {
-		let columns = {}
-		let rows = {}
-		for (let x = 0; x < this.width; x++)
-			for (let y = 0; y < this.height; y++) {
-				const index = x + this.width * y
-				if (chunk[index] !== this.last[index]) {
-					if (!columns.start) columns.start = x
-					if (!rows.start) rows.start = y
-					if (!rows.end || y > rows.end) rows.end = y
-					columns.end = x
-				}
-			}
-		if (columns.start) {
-			const column = columns.end - columns.start || 1 // if only one column updated then we take 1
-			const row = rows.end - rows.start || 1
-			const x = column.start
-			const y = rows.start
-			this.push({ column, row, x, y, chunk })
-		}
+	// buffer to id ndArray
+	async _transform(chunk, _, next) {
+		const { width, height, datas } = await fromImage(chunk.toString('utf8')) // TODO find which imput i should listen for
```

Comment:
currently the stream only take images path as input, i need to find an efficient way to extract frames of a streamed video (to avoid downloading the entire video first)